File: app/src/main/java/com/example/jetpackmvvm/sample/ui/switchsample/SwitchSampleFragment.java
package com.example.jetpackmvvm.sample.ui.switchsample;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.example.jetpackmvvm.sample.databinding.FragmentSwitchSampleBinding;
import com.example.jetpackmvvm.sample.util.MainViewModel;

public class SwitchSampleFragment extends Fragment {

    private FragmentSwitchSampleBinding mBinding;
    private MainViewModel mViewModel;

    @Override
    public void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        mBinding = FragmentSwitchSampleBinding.inflate(inflater, container, false);
        mBinding.setLifecycleOwner(this);
        mViewModel = new ViewModelProvider(requireActivity()).get(MainViewModel.class);
        mViewModel.getBannerState().observe(getViewLifecycleOwner(), state -> {
            mBinding.switch1.setChecked(state);
        });
        mBinding.switch1.setOnCheckedChangeListener((view, isChecked) -> {
            mViewModel.setBannerState(isChecked);
        });
        return mBinding.getRoot();
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        menu.clear();
    }
}

File: app/src/main/java/com/example/jetpackmvvm/sample/ui/switchsample/SwitchSampleFragment.kt
package com.example.jetpackmvvm.sample.ui.switchsample

import android.os.Bundle
import android.view.*
import android.widget.CompoundButton
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.jetpackmvvm.sample.databinding.FragmentSwitchSampleBinding
import com.example.jetpackmvvm.sample.util.MainViewModel

class SwitchSampleFragment : Fragment() {
    private lateinit var binding: FragmentSwitchSampleBinding
    private lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = FragmentSwitchSampleBinding.inflate(inflater, container, false)
        binding.lifecycleOwner = this
        viewModel = ViewModelProvider(requireActivity()).get(MainViewModel::class.java)
        viewModel.showBanner.observe(viewLifecycleOwner, { state: Boolean ->
            binding.switch1.isChecked = state
        })
        binding.switch1.setOnCheckedChangeListener { _, isChecked: Boolean ->
            viewModel.setBannerState(
                isChecked
            )
        }
        return binding.root
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        menu.clear()
    }
}